plugins/source/azure/resources/services/armcdn/endpoints_mock_test.go

```diff
@@ -0,0 +1,56 @@
+// Code generated by codegen; DO NOT EDIT.
+
+package armcdn
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cdn/armcdn"
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/faker"
+	"github.com/julienschmidt/httprouter"
+)
+
+func createEndpoints() (*client.Services, error) {
+	var item armcdn.EndpointsClientListByProfileResponse
+	if err := faker.FakeObject(&item); err != nil {
+		return nil, err
+	}
+	emptyStr := ""
+	item.NextLink = &emptyStr
+	mux := httprouter.New()
+	mux.GET("/*filepath", func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
```

Comment:
I personally don't see a lot of difference between the generated interface + mock on it vs transport-based mock tests, but prefer the interface+mocks better.